metrics/metrics.go

```diff
@@ -0,0 +1,135 @@
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+)
+
+const defaultEndpoint = "gradle-analytics.services.bitrise.io"
+const defaultPort = 443
+
+// Sync the major version of this step and the plugin.
+// Use the latest 1.x version of the plugin, so we don't have to update this definition after every plugin release.
+// But don't forget to update this to `2.+` if the library reaches version 2.0!
+const defaultPluginVersion = "main-SNAPSHOT" // TODO: change to 1.+
+
+type MetricsCollector struct {
+	envRepo      env.Repository
+	cmdFactory   command.Factory
+	pathProvider pathutil.PathProvider
+	logger       log.Logger
+	gradlewPath  string
+	buildFilePath string
+
```

Comment:
Do you have this empty line here on purpose?

Reply:
Somewhat 😆  The fields above this are constructor params, while the field below is the inner state of the struct, the value is stored there temporarily between two method calls.